src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
Use array methods instead of forEach, provide more descriptive names for params.
Note that If `people.find` does not find a person with the right name (i.e. there is no mother with that name in the people array), it will return an `undefined` value, therefore you need to create check statement:

Example of fixed code:
```javascript
  const ageDifferenceSum = children.reduce((accumulator, child) => {
    const mother = people.find(person => person.name === child.mother);
    return mother ? accumulator + (child.born - mother.born) : accumulator;
  }, 0);
  
  return ageDifferenceSum/children.length;
```